frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -3052,6 +3052,9 @@ const IR::Node *TypeInference::postorder(IR::PathExpression *expression) {
         if (type != nullptr)
             // may be nullptr because typechecking may have failed
             type = cloneWithFreshTypeVariables(type->to<IR::Type_MethodBase>());
+    } else if (decl->is<IR::Type_Declaration>()) {
```

Comment:
Exactly my thought. And a `PathExpression` is an expression, so it can be used to build more complex expressions. Having a type there directly does not make much sense.

Reply:
Any name that is in-scope as a type will be lexed as a `TYPE_IDENTIFIER` and recognized as a `typeName` in the parser, so should create a `Type_Name` rather than a `Path`.  The exception is when the type is used before its declaration, which (I think) can only happen with type parameters.